desktop-src/direct3dhlsl_current/hlsl-method-gatherCmp.md

```diff
@@ -0,0 +1,283 @@
+# GatherCmp Method
+
+Gets the four samples that would be used for bilinear interpolation when sampling a texture.
+
+Method Signatures vary per supported TextureObject type:
+
+### Signatures
+
+#### Texture2D
+```syntax
+// same as GatherRed
+ElementType Texture2D<ElementType>::GatherCmp(
+      in  SamplerState  samp,
+      in  float2        coord,
+      in  float         compareValue,
+      in  int2          offset,
+   [, out uint          status ] );
+
+ElementType Texture2D<ElementType>::GatherCmpRed(
+      in  SamplerState  samp,
+      in  float2        coord,
+      in  float         compareValue,
+      in  int2          offset,
+   [, in  int2          offset2]
+   [, in  int2          offset3]
+   [, in  int2          offset4]   
+   [, out uint          status ] );
+
+ElementType Texture2D<ElementType>::GatherCmpGreen(
```

Comment:
Do we need to expand out each signature for Red, Green, Blue, and Alpha?  This is just component selection on identical method overloads.  Could the method name be written: `GatherCmp[Red|Green|Blue|Alpha]` with one signature instead?  (same for normal Gather)

Reply:
I see what you are saying.  It seems like a good place to spell out everything is in the function signatures.  Of course, can easily correct if needed.